ShelterModule/Program.cs

```diff
@@ -45,18 +47,32 @@ private static void ApplyMigrations(IServiceProvider services)
         context.Database.Migrate();
     }
 
-    private static void ConfigureServices(IServiceCollection services, IConfiguration configuration)
+    private static void ConfigureServices(IServiceCollection services, IConfiguration configuration, bool isDevelopement)
     {
-        services.AddDbContext<PetShareDbContext>(options =>
-                                                     options.UseSqlServer(configuration.
-                                                                              GetConnectionString(PetShareDbContext.
-                                                                                                      DbConnectionStringName)));
+        if(isDevelopement)
+        {
+            services.AddDbContext<PetShareDbContext>(options =>
+                options.UseSqlServer(configuration.GetConnectionString(PetShareDbContext.DbConnectionStringName)));
+        }
+        else
+        {
+            services.AddDbContext<PetShareDbContext>(options =>
+                options.UseSqlServer(configuration.GetValue<string>("PetShareDbConnectionString")));
+        }
 
         services.AddScoped<IShelterQuery, ShelterQuery>();
         services.AddScoped<IShelterCommand, ShelterCommand>();
         services.AddScoped<IPetQuery, PetQuery>();
         services.AddScoped<IPetCommand, PetCommand>();
     }
 
-    private static void ConfigureOptions(IServiceCollection services) { }
+    private static void ConfigureOptions(WebApplicationBuilder builder) 
+    {
+        if (!builder.Environment.IsDevelopment())
+        {
+            var keyVaultUrl = new Uri(builder.Configuration.GetValue<string>("KeyVaultURL")!);
```

Comment:
We shouldn't use `!` here. `!` suppresses compiler warnings that the variable can be null, but it doesn't check/enforce this in any way. Since we need this value to not be `null` for the app to function, we should do `?? throw InvalidOperationException("message that tells us what's wrong")` instead (IDK if `InvalidOperationException` would be best here, just some exception).